cpp/src/gandiva/like_holder.cc

```diff
@@ -22,10 +22,20 @@
 #include "gandiva/regex_util.h"
 
 namespace gandiva {
+RE2 LikeHolder::starts_with_regex_(R"((\w|\s|\\-)*\.\*)");
+RE2 LikeHolder::ends_with_regex_(R"(\.\*(\w|\s|\\-)*)");
+RE2 LikeHolder::is_substr_regex_(R"(\.\*(\w|\s|\\-)*\.\*)");
 
-RE2 LikeHolder::starts_with_regex_(R"((\w|\s)*\.\*)");
-RE2 LikeHolder::ends_with_regex_(R"(\.\*(\w|\s)*)");
-RE2 LikeHolder::is_substr_regex_(R"(\.\*(\w|\s)*\.\*)");
+std::string& RemovePatternEscapeChars(const FunctionNode& node, std::string& pattern) {
+  if (node.children().size() != 2) {
```

Comment:
why is this required ? the pattern that is present here is the pcre pattern and any escapes should have already been removed as part of SqlLikePatternToPcre().

can you add a test which would fail without this change (RemovePatternEscapeChars part) and passes with the change ?

Reply:
There's already one(TestUtf8.TestLikeWithEscape), this is needed because when a % is escaped the SqlLikePatternToPcre keeps the escape char interfering on the string to search on starts_with expression.